packages/cli/generators/relation/index.js

```diff
@@ -338,50 +430,64 @@ module.exports = class RelationGenerator extends ArtifactGenerator {
 
   async promptRelationName() {
     if (this.shouldExit()) return false;
-    if (
-      this.artifactInfo.relationType === relationUtils.relationType.belongsTo
-    ) {
-      this.artifactInfo.destinationModelPrimaryKey = await relationUtils.getModelPrimaryKeyProperty(
-        this.fs,
-        this.artifactInfo.modelDir,
-        this.artifactInfo.destinationModel,
-      );
-      if (this.artifactInfo.destinationModelPrimaryKey == null) {
-        return this.exit(
-          new Error(ERROR_DESTINATION_MODEL_PRIMARY_KEY_DOES_NOT_EXIST),
-        );
-      }
-
-      this.artifactInfo.destinationModelPrimaryKeyType = relationUtils.getModelPropertyType(
-        this.artifactInfo.modelDir,
-        this.artifactInfo.destinationModel,
-        this.artifactInfo.destinationModelPrimaryKey,
-      );
-    }
-
     if (this.options.relationName) {
       debug(
         `Relation name received from command line: ${this.options.relationName}`,
       );
       this.artifactInfo.relationName = this.options.relationName;
     }
+    this.artifactInfo.defaultRelationName = this._getDefaultRelationName();
+    // for hasMany && hasOne, the source key is the same as the relation name
+    const msg =
+      this.artifactInfo.relationType === 'belongsTo'
```

Comment:
This property name will be the source key and the relation name for hasMany and hasOne. So I separate it from belongsTo here.